server.go

```diff
@@ -262,6 +263,7 @@ func (s *Server) handleGetMetadata(w *responseWirterWithStatus, r *http.Request)
 	if err := json.NewEncoder(w).Encode(gmr); err != nil {
 		log.Errorw("Failed to write get metadata response", "err", err, "key", sk)
 	}
+	w.WriteHeader(http.StatusOK)
```

Comment:
This is a footgun: according to go documentation, not calling `WriteHeader` means 200 will be written automatically. 

Instead of doing this, would it make sense to set the default value for `status` to `200` in the wrapper struct?